flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/BatchingStateChangeUploaderTest.java

```diff
@@ -94,14 +97,23 @@ public void testSizeThreshold() throws Exception {
     @Test
     public void testDelay() throws Exception {
         int delayMs = 50;
+        ManuallyTriggeredScheduledExecutorService scheduler =
+                new ManuallyTriggeredScheduledExecutorService();
         withStore(
                 delayMs,
                 Integer.MAX_VALUE,
+                scheduler,
                 (store, probe) -> {
+                    scheduler.triggerAll();
                     List<StateChangeSet> changeSets = getChanges(4);
                     upload(store, changeSets);
                     assertTrue(probe.getUploaded().isEmpty());
-                    Thread.sleep(delayMs * 2);
+                    assertTrue(
```

Comment:
Is this assert statemanet must exist here?

Reply:
The intention is to check that the delay, that it is equal to what was configured.